### ISSUE TITLE
New vue file will get `Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.` when typescript 3.9-4.0 in VSCode

### ISSUE THREAD

**Issue by yoyo930021** [typescript-eslint/typescript-eslint#2127]
This problem will effect typescript 3.9~4.0 .
Fixed in typescript 4.1 https://github.com/typescript-eslint/typescript-eslint/issues/2127#issuecomment-707426111

**Repro**
https://github.com/yoyo930021/typescript-eslint-vue-bug

```javascript
module.exports = {
  root: true,
  parser: 'vue-eslint-parser',
  env: {
    es6: true,
    node: true,
  },
  extends: ['plugin:vue/essential'],
  parserOptions: {
    // Local version of @typescript-eslint/parser
    parser: '@typescript-eslint/parser',
    project: './tsconfig.json',
    sourceType: 'module',
    extraFileExtensions: ['.vue'],
  },
  plugins: [
    // Local version of @typescript-eslint/eslint-plugin
    '@typescript-eslint',
  ]
};
```

```Vue
<template>
  <div></div>
</template>

<script>
export default {

}
</script>

<style>

</style>
```

**Expected Result**
All correct.
**Actual Result**
```
Parsing error: "parserOptions.project" has been set for @typescript-eslint/parser.
The file does not match your project config: src/App copy.vue.
The file must be included in at least one of the projects
```
**Additional Info**
1. open minimal reproduction repro in vscode.
2. copy App.vue to App copy.vue
3. Get error.

![May-29-2020 13-55-51](https://user-images.githubusercontent.com/9126398/83226256-267af900-a1b4-11ea-9217-f16abe0af92e.gif)

**Versions**

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/typescript-estree` | `3.0.2` |
| `TypeScript`                           | `3.9.3` |
| `node`                                 | `13.14.0` |
| `npm`                                  | `6.14.4` |

**Comment by yoyo930021** [comment]
I think about #1813, #1833.
Perhaps typescript 3.9 does not listen extra file extensions?

**Comment by bradzacher** [comment]
This will require some investigation - it should be work based on everything I understand.
I won't have bandwidth to look into this for some time, however.

**Comment by ericksonc** [comment]
FWIW, I often get this error, and then quitting and reopening VSCode makes it go away. It usually pops up again after a few hours, I have no idea what's happening to trigger it though.

**Comment by yoyo930021** [comment]
> FWIW, I often get this error (usually on existing Vue single file components I've had open in the editor for a while) and then quitting and reopening VSCode makes it go away. It usually pops up again after a few hours, and I have no idea what's happening to trigger it though.

You can downgrade TypeScript to 3.8 temporarily.

**Comment by sbillet** [comment]
The problem is still present with TypeScript 4.0.2 (2020-08-21) and 3.9.7 (2020-07-16). On the other hand, the older 3.8.3 (2020-02-28) works fine for me.

@sheetalkamat You made a fix in TS related to [1813](https://github.com/typescript-eslint/typescript-eslint/issues/1813). Do you expect the change to fix this issue? If yes, in which TS release version will it be included?

**Comment by bradzacher** [comment]
Vue is just not well supported by this project.

I don't have any experience in Vue, and I don't have the cycles to learn Vue and investigate whilst also maintaining everything else.

We'd love for someone from the Vue community to champion support.

Without a champion, this will likely remain broken.

**Comment by yoyo930021** [comment]
I studied it.

For the same reason I thought.
https://github.com/typescript-eslint/typescript-eslint/issues/2127#issuecomment-635776866

I open a issue and add a test in TypeScript project.
https://github.com/microsoft/TypeScript/issues/40495

**Comment by yoyo930021** [comment]
This problem will solved in TypeScript 4.1
https://github.com/microsoft/TypeScript/issues/40495#issuecomment-707262286

**Comment by yoyo930021** [comment]
I'm also interested in helping with questions about Vue, TypeScript, ESLint.
If you have a similar problem, you can tag me to help you with it.

**Comment by yoyo930021** [comment]
https://devblogs.microsoft.com/typescript/announcing-typescript-4-1/
